state-chain/pallets/cf-threshold-signature/src/lib.rs

```diff
@@ -1199,6 +1200,66 @@ pub mod pallet {
 	}
 }
 
+macro_rules! append_chain_to_name {
+	($name:ident) => {
+		match T::TargetChain::NAME {
+			"Ethereum" => concat!(stringify!($name), "Ethereum"),
+			"Polkadot" => concat!(stringify!($name), "Polkadot"),
+			"Bitcoin" => concat!(stringify!($name), "Bitcoin"),
+			"Arbitrum" => concat!(stringify!($name), "Arbitrum"),
+			"Solana" => concat!(stringify!($name), "Solana"),
+			_ => concat!(stringify!($name), "Other"),
+		}
```

Comment:
```suggestion
		concat!(stringify!($name), T::TargetChain::NAME)
```

But as mentioned elsewhere it will have to be ChainCrypto not Chain.